modules/core/src/smithy4s/http/internals/SchemaVisitorMetadataReader.scala

```diff
@@ -142,10 +142,19 @@ private[http] class SchemaVisitorMetadataReader()
       values: List[EnumValue[E]],
       total: E => EnumValue[E]
   ): MetaDecode[E] = {
-    MetaDecode
-      .from(
-        s"Enum[${values.map(_.stringValue).mkString(",")}]"
-      )(string => values.find(_.stringValue == string).map(_.value))
+    if (hints.get[IntEnum].isDefined) {
+      MetaDecode
+        .from(
+          s"Enum[${values.map(_.stringValue).mkString(",")}]"
```

Comment:
Not sure if we want to be using the `stringValue` in cases like this.. probably not, but that's what I had in the other PR. The ordinal would look a bit more confusing, but would be more relevant